packages/patternfly-3/patternfly-react/src/components/DateTimePicker/TimeComponents/PickTimeClock.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { noop } from '../../../common/helpers';
+
+class PickTimeClock extends React.Component {
+  state = {
+    ampm: this.props.time.getHours() >= 12 ? 'PM' : 'AM'
+  };
+  static getDerivedStateFromProps(nextProps, prevState) {
+    return {
+      ampm: nextProps.time.getHours() >= 12 ? 'PM' : 'AM'
+    };
+  }
+  setTime = (type, amount) => {
+    const { time } = this.props;
+    if (type === 'H') {
+      time.setHours(time.getHours() + amount);
+    } else if (type === 'M') {
+      time.setMinutes(time.getMinutes() + amount);
+    }
+    this.props.setSelected(time);
+  };
+  toggleAMPM = () => {
+    const { time } = this.props;
+    if (this.state.ampm === 'AM') {
+      time.setHours(time.getHours() + 12);
+      this.setState({ ampm: 'PM' });
+    } else {
+      time.setHours(time.getHours() - 12);
+      this.setState({ ampm: 'AM' });
+    }
+    this.props.setSelected(time);
+  };
+  render() {
+    const { time, toggleTimeTable } = this.props;
+    const minutes = time.getMinutes();
+    const hours = time.getHours() % 12 || 12;
+
+    return (
+      <div className="timepicker-picker">
+        <table>
+          <tr>
+            <td onClick={() => this.setTime('H', 1)}>
+              <a href="#" title="Increment Hour" className="btn clock-btn">
```

Comment:
because you used `href="#"`
when clicking on the time-picker buttons, it reloads the page or take you to another page:

![storybook 2](https://user-images.githubusercontent.com/26363699/53696488-03567380-3dd0-11e9-9ff8-8507416c398a.gif)